filecoin-proofs/tests/paramfetch/prompts_to_fetch.rs

```diff
@@ -0,0 +1,164 @@
+use std::collections::btree_map::BTreeMap;
+
+use failure::Error as FailureError;
+
+use crate::paramfetch::support::session::ParamFetchSessionBuilder;
+use crate::support::tmp_manifest;
+use blake2b_simd::State as Blake2b;
+use filecoin_proofs::param::ParameterData;
+use rand::Rng;
+
+/// Produce a random sequence of bytes and first 32 characters of hex encoded
+/// BLAKE2b checksum. This helper function must be kept up-to-date with the
+/// parampublish implementation.
+fn rand_bytes_with_blake2b() -> Result<(Vec<u8>, String), FailureError> {
+    let bytes = rand::thread_rng().gen::<[u8; 32]>();
+
+    let mut hasher = Blake2b::new();
+
+    let mut as_slice = &bytes[..];
+
+    std::io::copy(&mut as_slice, &mut hasher)?;
+
+    Ok((
+        bytes.iter().cloned().collect(),
+        hasher.finalize().to_hex()[..32].into(),
+    ))
+}
+
+#[test]
+fn nothing_to_fetch_if_cache_fully_hydrated() {
+    Ok::<(), FailureError>(())
+        .and_then(|_| {
```

Comment:
no need for this, you can return `Result` from tests in latest rust: https://doc.rust-lang.org/edition-guide/rust-2018/error-handling-and-panics/question-mark-in-main-and-tests.html

Reply:
Oh very nice!